src/core/nodeOps.ts

```diff
@@ -325,4 +347,19 @@ export const nodeOps: (context: TresContext) => RendererOptions<TresObject, Tres
     cloneNode: () => noop('cloneNode'),
     insertStaticContent: () => noop('insertStaticContent'),
   }
+
+  function prepare<T extends TresObject>(obj: T, state: Partial<LocalState>): TresInstance {
```

Comment:
Hey @alvarosabu 

>  Its always better practice to declare functions before calling them.

I usually prefer to let implementation functions be "hoisted", so that code can be written using this structure:

```md
* Important Fn 1 – calls Implementation Detail Fn A, B
* Important Fn 2 – calls Implementation Detail Fn B, C
* Implementation Detail Fn A
* Implementation Detail Fn B
* Implementation Detail Fn C
```

If we don't "hoist", then the implementation details go before the important stuff. 

```md
* Implementation Detail Fn A
* Implementation Detail Fn B
* Implementation Detail Fn C
* Important Fn 1 – calls Implementation Detail Fn A, B
* Important Fn 2 – calls Implementation Detail Fn B, C
```

In this case, this PR is one in a series. I'm working on the next one in the series and it adds another implementation function, which is rather long. That pushes the important functions, like `createElement` down pretty far in the page.

## Moving forward?

Maybe as a compromise, we move the implementation functions into `utils` and pass in `nodeOps`? What do you think?

Reply:
Agree, I think it makes total sense to move them to a utils folder, they could live in the same file or we can separate them by nature and export them all with a `utils/index.ts`, wdyt?